questionnaire/questionnaire-service.js

```diff
@@ -154,9 +164,107 @@ function createQuestionnaireService(spec) {
         return answerResource;
     }
 
+    async function retrieveCaseReferenceNumber(questionnaireId) {
+        // TODO: this is here temporarily until the message bus queue addition actually
+        // kicks things off properly behind the scenes. after that is the case
+        // then this comment and the line below (the return) can be remove.
+        // return '12\\123456';
+        const result = await getQuestionnaire(questionnaireId);
+        const {questionnaire} = result.rows && result.rows[0];
+        const caseReference =
+            questionnaire.answers &&
+            questionnaire.answers.system &&
+            questionnaire.answers.system['case-reference'];
+        if (caseReference) {
+            return caseReference;
+        }
+
+        return null;
+    }
+
+    async function startSubmission(questionnaireId) {
+        await updateQuestionnaireSubmissionStatus(questionnaireId, 'IN_PROGRESS');
+        const messageBus = createMessageBusCaller({logger});
+        const submissionResponse = await messageBus.post('submissionQueue', {
+            applicationId: questionnaireId
+        });
+        if (
+            !submissionResponse ||
+            !submissionResponse.body ||
+            submissionResponse.body !== 'Message sent'
+        ) {
+            await updateQuestionnaireSubmissionStatus(questionnaireId, 'FAILED');
+            await db.createQuestionnaireSubmission(questionnaireId, 'FAILED');
+        }
+    }
+
+    async function getSubmissionResponseData({questionnaireId, submissionStatus = undefined}) {
+        const response = {
+            data: {
+                type: 'submissions',
+                attributes: {
+                    questionnaireId,
+                    submitted: false,
+                    status: submissionStatus,
+                    caseReferenceNumber: null,
+                    applicantEmail: null,
+                    applicantName: null
+                }
+            }
+        };
+
+        // if the case reference number is in the database it means that the questionnaire
+        // has been submitted.
+        const caseReferenceNumber = await retrieveCaseReferenceNumber(questionnaireId);
+
+        if (caseReferenceNumber !== null) {
+            response.data.attributes.submitted = true;
+        }
+        if (response.data.attributes.submitted) {
```

Comment:
can you elaborate please?

Reply:
I was just speaking to @BarryPiccinni  about this...

What about the questionnaire template having some sort of onComplete key e.g.

```javascript
{
    onComplete: {
        emailTemplateId: "some-notify-template-name",
        emailTemplatePlaceholderMap: {
            "applicantName": "p-applicant-name",
            ...
        }
    }
}
```

When the consumer then polls the submission status, if the DCS returns "complete" (or whatever it is), it then knows to kick off the sending of the confirmation email? This would mean the template dictates what it should do.

Reply:
the above example has "p-applicant-name" but hopefully you get what I mean, some mapping to the correct value

Reply:
where about in the template were you thinking that this `onComplete` key would live?

For the `applicantName` stuff, are you saying we could have some sort of mapping (e.g. `p-what-is-your-name.q-what-is-your-name`) that represents the location of the value we want to get?

Reply:
I'm not sure {meta: {onComplete:...}}, have a think and see what makes sense to you.

Yeah the "emailTemplatePlaceholderMap" would map a placeholder (think that's the notify terminology) to an answer. I'd use some standard such as a [json pointer](https://tools.ietf.org/html/rfc6901)